packages/scanner/src/Scanner.ts

```diff
@@ -0,0 +1,225 @@
+    return this.rpcProvider.send('state_getRuntimeVersion', [blockHash]);
+  }
+
+  public async getBlockHash(at: number | Hash): Promise<Hash> {
+    if (typeof at === 'number' && !isNaN(at as any) && !isHex(at)) {
+      const blockHash = this.rpcProvider.send('chain_getBlockHash', [at]);
+      return blockHash;
+    } else {
+      return at as Hash;
+    }
+  }
+
+  public async getBlockAt(blockAt?: { blockHash?: Hash; blockNumber?: number }): Promise<BlockAt> {
+    if (blockAt?.blockHash && blockAt?.blockNumber !== undefined) {
+      return {
+        blockHash: blockAt.blockHash,
+        blockNumber: blockAt.blockNumber
+      };
+    }
+    if (!blockAt) {
+      const header = await this.rpcProvider.send('chain_getHeader', []);
+      const blockNumber = Number(header.number);
+      const blockHash = await this.rpcProvider.send('chain_getBlockHash', []);
+      return {
+        blockNumber: blockNumber,
+        blockHash: blockHash
+      };
+    } else if (blockAt.blockNumber !== undefined) {
+      const blockHash = await this.rpcProvider.send('chain_getBlockHash', [blockAt.blockNumber]);
+      return {
+        blockNumber: blockAt.blockNumber,
+        blockHash: blockHash
+      };
+    } else if (blockAt.blockHash) {
+      const header = await this.rpcProvider.send('chain_getHeader', [blockAt.blockHash]);
+      return {
+        blockNumber: Number(header.number),
+        blockHash: blockAt.blockHash
+      };
+    } else {
+      throw new Error('expect blockHash or blockNumber');
+    }
+  }
+
+  public async getChainInfo(_blockAt?: BlockAtOptions): Promise<ChainInfo> {
+    const { blockHash, blockNumber } = await this.getBlockAt(_blockAt);
+    const runtimeVersion = await this.getRuntimeVersion(blockHash);
+    const cacheKey = `${runtimeVersion.specName}-${runtimeVersion.specVersion}`;
+    const registry = new TypeRegistry();
+    if (!this.chainInfo[cacheKey]) {
+      const rpcdata: string = await this.rpcProvider.send('state_getMetadata', [blockHash]);
+      this.chainInfo[cacheKey] = {
+        min: blockNumber,
+        max: blockNumber,
+        bytes: rpcdata,
+        metadata: new Metadata(registry, rpcdata),
+        registry: registry,
+        runtimeVersion: runtimeVersion
+      };
+    } else {
+      this.chainInfo[cacheKey].min = Math.min(this.chainInfo[cacheKey].min || Number.MAX_SAFE_INTEGER, blockNumber);
+      this.chainInfo[cacheKey].max = Math.max(this.chainInfo[cacheKey].max || Number.MIN_SAFE_INTEGER, blockNumber);
+    }
+    return this.chainInfo[cacheKey];
+  }
+
+  public async getEvents(_blockAt: BlockAtOptions): Promise<Vec<EventRecord>> {
+    const blockAt = await this.getBlockAt(_blockAt);
+    const { metadata, registry } = await this.getChainInfo(blockAt);
+    const eventsStorageKey = new StorageKey(registry, metadata.query.system.events);
+    const raw: Hash = await this.rpcProvider.send('state_getStorage', [eventsStorageKey.toHex(), blockAt.blockHash]);
```

Comment:
why `raw: Hash`? shouldn't it be `Bytes`?

Reply:
This is just an alias for string, so I'll replace all the Hash with Bytes

Reply:
oh so it not the `Hash` type from polkadot.js